src/code_43F0.c

```diff
@@ -242,7 +263,32 @@ s32 round(f32 arg0) {
     }
 }
 
-INCLUDE_ASM(f32, "code_43F0", clamp_angle, f32 theta);
+f32 clamp_angle(f32 theta) {
+    if (fabsf(theta) > 3600.0f) {
+        s32 thetaInt = theta;
+
+        if (thetaInt >= 0) {
+            thetaInt = -(thetaInt / 360 * 360) + thetaInt;
+        }
+        else {
+            thetaInt = -thetaInt;
+            thetaInt = (thetaInt / 360 * 360) - thetaInt;
+        }
+
+        return thetaInt;
+    }
+
+    do {
+        if (theta < 0.0f) {
+            theta += 360.0f;
+        }
+        if (theta >= 360.0f) {
+            theta -= 360.0f;
+        }
+    } while (!(theta >= 0.0f) || !(theta < 360.0f));
```

Comment:
Does this match if you remove the `!`s and use `&&` instead

Reply:
nary